docs/architecture/devops-for-aspnet-developers/actions-vs-pipelines.md

```diff
@@ -0,0 +1,118 @@
+> It's possible to install tools and SDKs when running pipelines on hosted agents. If the install steps don't take long, this is viable. However, if the tools/software take a long time to install, then you may be better off with a private agent or self-hosted runner, since the install steps will need to execute for every run of the workflow.
+
+### Azure DevOps agents
+
+Every Azure DevOps account has a hosted pool with a single agent that can run one job at a time and some amount of free build minutes. You can purchase additional "hosted pipelines" in Azure DevOps. When you purchase an additional hosted pipeline, you're really removing the build minutes limit and adding _concurrency_: one pipeline can run one job at a time. Two pipelines can run two jobs simultaneously, and so on.
+
+### Comparison of agents
+
+Feature|GitHub|Azure Pipelines|Links
+--|--|--|--
+Hosted agents for public repos/projects|Free|[No free minutes](https://devblogs.microsoft.com/devops/change-in-azure-pipelines-grant-for-public-projects/) for public projects|[Azure Pipelines](/azure/devops/pipelines/agents/hosted?view=azure-devops&tabs=yaml#capabilities-and-limitations) [GitHub](https://github.com/features/actions)
+Hosted agents for private repos/projects|2,000 minutes free per month, 3,000 minutes for Pro and Team licenses, 50,000 minutes for Enterprise license. Additional minutes may be purchased.|One free parallel job that can run for up to 60 minutes each time, until you've used 1,800 minutes (30 hours) per month. You can pay for additional capacity per parallel job. Paid parallel jobs remove the monthly time limit and allow you to run each job for up to 360 minutes (6 hours).
+Cross-platform|Yes|Yes|
+Scale set agents|No|Yes|Read about scale-set agents [here](/azure/devops/pipelines/agents/scale-set-agents?view=azure-devops)
+
+## Comparison of GitHub Actions and Azure Pipelines
+
+Azure Pipelines (YAML pipelines) provide a mature set of features such as approvals, artifact storage, deployment jobs, environments, gates, stages, templates, triggers, variable groups and more. For a full list of Azure Pipelines features, refer to the table [here](/azure/devops/pipelines/get-started/pipelines-get-started?view=azure-devops#feature-availability).
+
+GitHub Actions are evolving rapidly and provide a set of features such as triggers for almost all GitHub events, artifact storage, environments and environment rules, starter templates, matrices and more. Read more about the full feature set of GitHub Actions [here](https://docs.github.com/actions).
+
+### Feature comparison
+
+(As of March 2021)
+
+Feature|Description|GitHub Actions|Azure Pipelines
+--|--|--|--
+Approvals|Define approval conditions before moving further in the pipeline|Yes|Yes
+Artifacts|Upload, store and download artifacts from jobs|Yes|Yes
+Caching|Cache folders or files for subsequent runs|Yes|Yes
+Conditions|Specify conditions for steps or jobs|Yes|Yes
+Container Jobs|Run jobs inside a container|No|Yes
```

Comment:
```suggestion
Container Jobs|Run jobs inside a container|Yes|Yes
```

GitHub Actions support either JavaScript or Container actions, is this not a container job? Individual actions are steps within a job, and runnable as a container.